firedrake/assemble.py

```diff
@@ -20,7 +20,7 @@
 
 
 def assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
-             inverse=False, nest=None):
+             inverse=False, nest=None, matfree=False, extra_ctx={}):
```

Comment:
I'm looking at it, with the idea of mat_type='aij' | 'nest' | 'matfree' and
no kwarg for nest or matfree in calls to assemble.  Similarly, the "nest"
kwarg in (N)LVP then will have to come in to the solver via the parameter
dictionary/options database.

The change to assemble.py looks pretty simple, but then we have to
propogate through all of the examples and tests.  Again, straightforward
but more time-consuming.

FWIW, I think this is a feature -- nesting is not properly part of the
problem description like say, UFL is -- it's part of the description of how
we solve the problem.

However, There is an issue here in that this will break a top-level
interface when merged to master, albeit with a very minor patch-up in the
user code.  That's worth some discussion, at least acknowledging a minor,
1-time inconvenience to users.




On Mon, Aug 15, 2016 at 10:14 AM, Lawrence Mitchell <
notifications@github.com> wrote:

> In firedrake/assemble.py
> <https://github.com/firedrakeproject/firedrake/pull/853#discussion_r74777104>
> :
>
> > @@ -20,7 +20,7 @@
> >
> >
> >  def assemble(f, tensor=None, bcs=None, form_compiler_parameters=None,
> > -             inverse=False, nest=None):
> > +             inverse=False, nest=None, matfree=False, extra_ctx={}):
>
> I will not get to it today now, go ahead. If you don't do it, I will go
> tomorrow.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/firedrakeproject/firedrake/pull/853/files/ff5ec4e0eabb94576ac69c9c82932eaa50b2d14e#r74777104>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADl6LxPSZXn_dutwSsCCyHCk2GmDOpdsks5qgIJBgaJpZM4JiV-g>
> .
>